crypto/CMakeLists.txt

```diff
@@ -1,3 +1,47 @@
+
+# Function to handle assembly files for ARM64 targets when using MSBuild (the "Visual Studio" generator)
+# This function is necessary because MSBuild ignores ARM64 assembly file dependencies
+function(msbuild_aarch64_asm)
+  set(options "")
+  set(oneValueArgs TARGET OUTPUT_OBJECTS)
+  set(multiValueArgs ASM_FILES)
+  
+  cmake_parse_arguments(MSBUILD_AARCH64_ASM "${options}" "${oneValueArgs}" "${multiValueArgs}" ${ARGV})
```

Comment:
TIL about cmake_parse_arguments, very neat.

I think cmake_parse_arguments was only added in CMake 3.5. Based on our CI we test back to 3.2 on Linux. Is it possible to implement msbuild_aarch64_asm without using cmake_parse_arguments?

Reply:
It was available in older CMake versions, it just required including the [CMakeParseArguments](https://cmake.org/cmake/help/latest/module/CMakeParseArguments.html#module:CMakeParseArguments) module prior to 3.5.

Reply:
Is that included automatically? Or something you can add?

Reply:
Yeah, I'll add it.